dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -372,20 +391,36 @@ rule tckgen:
     threads: 32
     resources:
         mem_mb=128000,
-        time=60*24,
+        time=60 * 24,
     log:
         f"{config['output_dir']}/logs/mrtrix/sub-{{subject}}/tckgen.log",
     container:
         config["singularity"]["mrtrix"]
     shell:
-        "tckgen -nthreads {threads} -algorithm iFOD2 -step {params.step} -select {params.sl} -exclude {input.cortical_ribbon} -exclude {input.convex_hull} -include {input.subcortical_seg} -mask {input.mask} -seed_image {input.mask} {input.fod} {output.tck} &> {log}"
+        "mkdir -p {params.tmp_dir} && "
+        "tckgen -nthreads {threads} -algorithm iFOD2 -step {params.step} -select {params.sl} -exclude {input.cortical_ribbon} -exclude {input.convex_hull} -include {input.subcortical_seg} -mask {input.mask} -seed_image {input.mask} {input.fod} {params.tmp_tck} &> {log} && "
+        "rsync -v {params.tmp_tck} {output.tck} >> {log} 2>&1"
 
 
 rule tcksift2:
     # Smith, R. E.; Tournier, J.-D.; Calamante, F. & Connelly, A. The effects of SIFT on the reproducibility and biological accuracy of the structural connectome. NeuroImage, 2015, 104, 253-265
     input:
         tck=rules.tckgen.output.tck,
         fod=rules.mtnormalise.output.wm_fod,
+    params:
```

Comment:
As above, I think `tmp_dir` and `tmp_weights` should be resources.

Reply:
Realized these aren't actually used here so will be removing this.